sw/device/lib/base/hardened_memory.c

```diff
@@ -78,11 +78,7 @@ void hardened_memcpy(uint32_t *restrict dest, const uint32_t *restrict src,
   HARDENED_CHECK_EQ(count, expected_count);
 }
 
-// The source of randomness for shred, which may be replaced at link-time.
-OT_WEAK
-uint32_t hardened_memshred_random_word(void) { return 0xcaffe17e; }
-
-void hardened_memshred(uint32_t *dest, size_t word_len) {
+void hardened_memshred(uint32_t (*rnd)(void), uint32_t *dest, size_t word_len) {
```

Comment:
Sorry - I'm late to the party and I missed the discussion at the SW-WG about this.

What is the issue with the existing solution: the weak symbol `hardened_memshread_random_word`?

This, to me, seems like the easiest solution:
- There is a sensible default when you naively use `memshred`.
- If your application has specific requirements about the randomness used to "shred" memory, then you can supply a strong symbol which will override the weak symbol.
- You don't need to plumb a new function argument through the codebase and/or your application to call `memsrhed`.
- You don't need to worry about whether having dynamic dispatch through a function pointer is something that and adversary can fault and cause a problem.

Reply:
+1